libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -81,10 +81,23 @@ void AP_Camera::take_picture()
 {
     WITH_SEMAPHORE(_rsem);
 
-    if (primary == nullptr) {
+    // call for each instance
+    for (uint8_t i = 0; i < AP_CAMERA_MAX_INSTANCES; i++) {
+        if (_backends[i] != nullptr) {
+            _backends[i]->take_picture();
+        }
+    }
+}
+
+void AP_Camera::take_picture(uint8_t instance)
+{
+    WITH_SEMAPHORE(_rsem);
+
+    auto *backend = get_instance(instance-1);
```

Comment:
I think we should remove this -1.  Everywhere else in AP the instance is zero based internally so we shouldn't change that unless absolutely necessary.